content/blog/debugging-nodejs-programs-using-chrome/index.md

```diff
@@ -0,0 +1,308 @@
+
+![Showing the asterisk once changes made but not saved](./edited-but-not-saved.png)
+
+In order to make your changes persist, you'll need to press `Ctrl + S` or `Command + S` in order to save the file (much like a Word document). Doing so will bring up a yellow triangle instead of an asterisk indicating _your changes are not saved to the source file but your changes will now take effect_. Re-running the `localhost:3000` route will now correct the behavior you're wanting, but if you open `app.js` in a program like Visual Studio Code, it will show the older broken code.
+
+![A screenshot showing the yellow triangle once this occurs](./temporarily-saved.png)
+
+Not only does VS Code not recognize your changes, but once you close your debugging window, you won't know what you'd changed in order to get your code to work. While this may help in short debugging sessions, this won't do for a longer session of code changes. To do that, you'll want your changes to save to the local file system.
+
+## Persisting Changes {#chrome-as-ide-persist-changes}
+
+In order to save the changes from inside the Chrome to the file system, you need to permit Chrome access to read and write your files. To do this, you'll want to press the "Add folder to workspace" button off to the left of the code screen.
+
+![A red rectangle around "Add folder to workspace" button](./add-folder-to-workspace.png)
+
+Selecting the folder your `app.js` is present in will bring up the dialog to give Chrome permission to view the files and folders within. You'll need to press "Allow" in order to save your saves to your file system
+
+![A screenshot showing the "Allow"/"Deny" dialog](./allow-fs-usage.png)
+
+Once done, you should now see a list of the files and folders in the parent folder. It should automatically have highlights over the `app.js` file and remove the yellow triangle in favor of another asterisk.
+
+![A screenshot as described in the previous paragraph](./fs-permitted-not-saved.png)
+
+
+
+As I'm sure you've guessed, the asterisk indicates that you'll need to save the file again. Once done (using the key combo), the asterisk should disappear.
+
+It's not just JavaScript files you're able to edit, though! You can click or use your keyboard to navigate the file tree of the parent folder. Doing so will allow you to edit and save changes to _any_ file in the filesystem. This would include the `package.json` file in the folder 
```

Comment:
```suggestion
It's not just JavaScript files you're able to edit, though! You can click or use your keyboard to navigate the file tree of the parent folder. Doing so will allow you to edit and save changes to _any_ file in the filesystem. This would include the `package.json` file in the folder.
```